enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotAllocation.scala

```diff
@@ -135,6 +144,43 @@ object SlotAllocation {
     PhysicalPlan(allocations.toMap, arguments.toMap)
   }
 
+  // NOTE: If we find a NestedPlanExpression within the given LogicalPlan, the slotConfigurations and argumentSizes maps will be updated
+  private def allocateExpressions(lp: LogicalPlan, nullable: Boolean, slots: SlotConfiguration,
+                                  slotConfigurations: mutable.Map[LogicalPlanId, SlotConfiguration],
+                                  argumentSizes: mutable.Map[LogicalPlanId, Size]): SlotConfiguration = {
+    val newSlots = lp.treeFold(slots) {
+      case p: LogicalPlan if p.assignedId != lp.assignedId =>
+        slots => (slots, None)
+
+      case e: ScopeExpression =>
+        slots => {
+          e.introducedVariables.foreach {
+            case v@Variable(name) =>
+              slots.newReference(name, true, CTAny)
```

Comment:
I'm still not convinced that we should allocate local variables among the streamed slots, but if you think it's the way to go then let's try.